### ISSUE TITLE
5.4.0 produces "no-changed-when" warnings

### ISSUE THREAD

**Issue by asafs932** [ansible-community/ansible-lint#1962]
##### Summary

i saw that using 5.4.0 produces warnings :
"no-changed-when: Commands should not change things if nothing needs doing"
while running with 5.3.2 everything is fine.

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details
5.4.0

##### OS / ENVIRONMENT

Linux

##### STEPS TO REPRODUCE

actual code (task) for example:
```
    - name: Install ansible-datadog role
      command: "ansible-galaxy install --force -r {{ ansible_home }}/datadog_role_reqs.yml"
```

output:
```
WARNING  Listing 15 violation(s) that are fatal
no-changed-when: Commands should not change things if nothing needs doing
tasks/datadog.yml:18 Task/Handler: Download ansible-datadog role

no-changed-when: Commands should not change things if nothing needs doing
tasks/datadog.yml:34 Task/Handler: Install ansible-datadog role
...
...

```
##### Desired Behavior

running ansible-lint shouldnt raise these warnings

Possible security bugs should be reported via email to `security@ansible.com`

##### Actual Behavior

produces warnings:
no-changed-when: Commands should not change things if nothing needs doing

**Comment by ssbarnea** [comment]
PR welcomed!

**Comment by ssbarnea** [comment]
Now that I look back, that is not really a bug, in fact it is result of improvement of these checks. You can always expect to see changes between versions, even between minor version changes.
